sec2_the-ruby-programming-language/2.8_hashes.rb

```diff
@@ -0,0 +1,50 @@
+sample_hash = {'a' => 1, 'b' => 2, 'c' => 3}
+my_details = {'name' => 'michelle', 'favcolor' => 'black'}
+p my_details['favcolor']  # "black"
+p sample_hash['b']        # 2
+p sample_hash             # {"a"=>1, "b"=>2, "c"=>3}
+
+another_hash = {a: 1, b: 2, c: 3}
```

Comment:
1. Whats the difference between doing hash in line 7 & 2 ?
2. When should you choose one of the other?

Reply:
1. The one in line 2 is using String as key while the one in line 7 uses Symbol as key (Symbol is immutable string)
2. If using the text contents itself is important use String. If identify the object itself is important use Symbol.

---
Question
1. When does text contents itself is important?

Reply:
I won't answer your question cause your answer is not accurate and it seems you copy paste from a source without doing proper research to understand what is a symbol and why use symbol in hash

Reply:
1. The one in line 2 is using String as key while the one in line 7 is using Symbol as key. Symbols are unique, so no matter how many times you use Symbols, it refers to the same address. Strings are not unique, for example, when you create `"process"` and `"process"`, both of these Strings although looks the same but they refer to different address. Symbols can save more memory space because it is unique no matter how many times you create it and it can be accessed faster than Strings because of its uniqueness.

2.  Use Symbol when you need uniqueness in representing something, so better use Symbol in hash. Use Strings for something that uniqueness is not mandatory.

Reply:
1. Good
2. Not 100% accurate but can.